Added Shopping Cart

NadavLedany opened the pull request.

NadavLedany committed:
Added first Shopping Cart screens and CartController
Added Cart FXML files.
dashboard/Controller was modified as needed
Login_screen refresh is fixed.

NadavLedany committed:
Added first Shopping Cart screens and CartController
Added Cart FXML files.
dashboard/Controller was modified as needed
Login_screen refresh is fixed.

lielfr commented on .idea/modules.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Why do we need this file?

lielfr replied:
Please remove it. We don't really need it and you also didn't upload the Order model.

lielfr commented on lilach-client/src/main/java/org/cshaifasweng/winter/CartController.java:
```diff
@@ -0,0 +1,31 @@
+import java.io.IOException;
+
+
+public class CartController extends PrimaryController implements Refreshable {
```
Do NOT extend PrimaryController

lielfr commented on lilach-client/src/main/java/org/cshaifasweng/winter/models/Cart.java:
```diff
@@ -0,0 +1,15 @@
+
+import java.util.List;
+
+public class Cart {
```
Generally speaking, we don't have models for aggregations of other models unless we need to add more information about them.

lielfr replied:
This is not an excuse. You can just check if the list is empty.I don't think we really need a Cart entity, and unless you convince me otherwise I want you to remove it.

lielfr commented on lilach-client/src/main/java/org/cshaifasweng/winter/DashboardController.java:
```diff
@@ -51,4 +47,23 @@ void handleComplaint(ActionEvent event) {
         setPage("complaint_handling");
     }
 
+    public boolean CartIsEmpty=true;   // to setup
```
Please move those declarations before all the methods. Also, why are they public?

lielfr committed:
Merge branch 'master' into nadav-master

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/dashboard.fxml:
```diff
@@ -16,6 +13,7 @@
             <Button layoutX="10.0" layoutY="64.0" maxHeight="1.7976931348623157E308" maxWidth="1.7976931348623157E308" mnemonicParsing="false" text="MenuItem4" />
             <Button layoutX="10.0" layoutY="91.0" maxHeight="1.7976931348623157E308" maxWidth="1.7976931348623157E308" mnemonicParsing="false" text="MenuItem5" />
             <Button layoutX="10.0" layoutY="114.0" maxHeight="1.7976931348623157E308" maxWidth="1.7976931348623157E308" mnemonicParsing="false" text="MenuItem5" />
+            <Button mnemonicParsing="false" onAction="#CartButton" prefHeight="46.0" prefWidth="69.0" text="Cart" />
```
The button doesn't fit here (see screenshot). Make it the same shape as the other buttons.
<img width="123" alt="Screen Shot 2020-01-19 at 11 53 19" src="https://user-images.githubusercontent.com/360928/72678833-4ab22500-3ab2-11ea-9214-db985be41b81.png">

lielfr commented on lilach-client/src/main/java/org/cshaifasweng/winter/LoginScreen.java:
```diff
@@ -1,4 +1,9 @@
 package org.cshaifasweng.winter;
 
-public class LoginScreen {
+public class LoginScreen implements Refreshable {
```
Change its name (Refactor->Rename) to LoginScreenController.

lielfr commented on lilach-client/src/main/java/org/cshaifasweng/winter/CartController.java:
```diff
@@ -0,0 +1,31 @@
+
+
+    @FXML
+    private void switchToDashboard() throws IOException {
```
Method name here is confusing. Please change it to something like "Back to Catalog".

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/Empty_Cart_Customer.fxml:
```diff
@@ -0,0 +1,23 @@
+<?import javafx.scene.layout.*?>
+<?import javafx.scene.text.*?>
+
+<AnchorPane prefHeight="768.0" prefWidth="904.0" xmlns="http://javafx.com/javafx/10.0.2-internal" xmlns:fx="http://javafx.com/fxml/1" fx:controller="org.cshaifasweng.winter.CartController">
```
Please reduce the height a bit, so it does not require scrolling.

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/Empty_Cart_Guest.fxml:
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This is nice, but try to reduce the prefHeight of the anchor pane a bit, or maybe just use the computed size there.

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/dashboard.fxml:
```diff
@@ -1,12 +1,9 @@
+<?import javafx.scene.control.*?>
+<?import javafx.scene.layout.*?>
 
-<HBox maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="768.0" prefWidth="1024.0" xmlns="http://javafx.com/javafx/11.0.1" xmlns:fx="http://javafx.com/fxml/1" fx:controller="org.cshaifasweng.winter.DashboardController">
```
Is your SceneBuilder version 11? If not, please update. If it is, then just make sure not to change that XML namespace. Please change it manually back to what it was.

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/Customer_Cart.fxml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Well, this is just the beginning, but that's better than nothing. Well done.

lielfr replied:
Please upload a screenshot.

lielfr commented on lilach-server/src/main/resources/org/cshaifasweng/winter/application.properties:
```diff
@@ -2,7 +2,7 @@
 # TODO: Change this once we have a proper DB initialization process
 spring.jpa.hibernate.ddl-auto=create-drop
-spring.datasource.url=jdbc:mysql://localhost:3306/lilach
+spring.datasource.url=jdbc:mysql://localhost:3306/lilach?serverTimezone=UTC
```
That's what we did together. Since it works on my machine as well, let's keep it.

lielfr reviewed (changes requested):
Please read my comments and change accordingly if necessary. I'll review it again today at 10PM.

NadavLedany reviewed (commented):
תודה, 
אני ממשיך לעבוד על מסך ההזמנה

NadavLedany committed:
Fixes for last commit
Added Order model, OrdercController

updated CartController to reach future order screen

lielfr commented on .idea/compiler.xml:
```diff
@@ -15,6 +15,7 @@
   </component>
   <component name="JavacSettings">
     <option name="ADDITIONAL_OPTIONS_OVERRIDE">
+      <module name="lilach-client" options="--add-exports=jdk.jfr/jdk.jfr.internal=ALL-UNNAMED" />
```
What is that?

NadavLedany replied:
סומן בטעות כנראה, כי סימנתי תפריט אב  שזה היה בתוכו

lielfr commented on lilach-client/pom.xml:
```diff
@@ -63,6 +63,18 @@
             <artifactId>eventbus</artifactId>
             <version>3.1.1</version>
         </dependency>
+        <dependency>
```
Why?

NadavLedany replied:
כנ"ל

lielfr commented on lilach-client/src/main/java/org/cshaifasweng/winter/CartController.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    @FXML
+    private void BackToCatalog() throws IOException {
```
Please change the first letter to be lowercase, e.g. backToCatalog

lielfr replied:
Change that as well.

NadavLedany replied:
אוקיי

NadavLedany replied:
I will change the Back to back...
what is **that**?

NadavLedany replied:
Done

lielfr replied:
Please commit and push.

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/Empty_Cart_Customer.fxml:
```diff
@@ -0,0 +1,23 @@
+
+<AnchorPane prefHeight="400.0" prefWidth="904.0" style="-fx-background-color: #FFFFFF;" xmlns="http://javafx.com/javafx/10.0.2-internal" xmlns:fx="http://javafx.com/fxml/1" fx:controller="org.cshaifasweng.winter.CartController">
+   <children>
+      <TextField layoutY="20.0" prefHeight="77.0" prefWidth="904.0" style="-fx-background-color: #FFCCFF;" text="Your cart is empty">
```
I see styling here. Why?

NadavLedany replied:
will be removed.
it was back to life after control+z.

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/Empty_Cart_Guest.fxml:
```diff
@@ -0,0 +1,34 @@
+         <font>
+            <Font size="14.0" />
+         </font></Button>
+      <TextField layoutY="20.0" prefHeight="77.0" prefWidth="904.0" style="-fx-background-color: #FFCCFF;" text="Your cart is empty">
```
Same here - why style it? Can you post a screenshot?

NadavLedany replied:
same

lielfr committed:
Merge branch 'master' of github.com:NadavLedany/lilach into nadav-master

lielfr committed:
Merge branch 'master' into nadav-master

# Conflicts:
#	lilach-client/src/main/java/org/cshaifasweng/winter/DashboardController.java
#	lilach-client/src/main/java/org/cshaifasweng/winter/LoginScreenController.java
#	lilach-client/src/main/resources/org/cshaifasweng/winter/dashboard.fxml
#	lilach-client/src/main/resources/org/cshaifasweng/winter/login_screen.fxml

lielfr committed:
Merge Resolve

lielfr committed:
pom.xml rollback

NadavLedany committed:
Removed the color from cart layout
added 3 order .fxml files(basic connections only).
added orderController
updated cart,order models.
updated Customer cart.

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/Customer_Cart.fxml:
```diff
@@ -0,0 +1,26 @@
+            <opaqueInsets>
+                <Insets top="1.0" />
+            </opaqueInsets></TextField>
+      <Button layoutX="670.0" layoutY="348.0" mnemonicParsing="false" onAction="#GoToOrder" prefHeight="50.0" prefWidth="104.0" text="Check Out">
```
Once again, pay attention to the first letter - should not be capital.

NadavLedany replied:
i will decide about cart model today.
will change to  goToOrder

lielfr commented on lilach-client/src/main/resources/org/cshaifasweng/winter/order_completed.fxml:
```diff
@@ -0,0 +1,24 @@
+
+<AnchorPane maxHeight="-Infinity" maxWidth="-Infinity" prefHeight="500.0" prefWidth="904.0" xmlns="http://javafx.com/javafx/10.0.2-internal" xmlns:fx="http://javafx.com/fxml/1" fx:controller="org.cshaifasweng.winter.OrderController">
+    <children>
+        <Label layoutX="46.0" layoutY="14.0" prefHeight="35.0" prefWidth="189.0" text="Thank You !">
```
Remove the space before the exclamation mark.

NadavLedany replied:
ok

lielfr reviewed (changes requested):
A few more things.

NadavLedany reviewed (commented):
Fixed according to review.

NadavLedany committed:
Fixes what needed according to last review,
among others:
Removed the styles from all .fxml's.
Deleted cart model.
Corrected first letter of methods to small letter in CartController, and dashboardController.
Fixed another warning before commit, will attach screenshot.

lielfr reviewed (approved):
Well done.
